infrastructure/tooling/src/generate/generators/go-version.js

```diff
@@ -67,15 +70,24 @@ exports.tasks = [{
         `MIN_GO_MINOR_VERSION=${goVersionMinor}`));
 
     utils.status({ message: 'workers/generic-worker/gw-decision-task/tasks.yml' });
+    const previousGoVersion = (await readRepoFile('workers/generic-worker/gw-decision-task/tasks.yml')).match(/go[0-9.]+/)[0].substring(2).split('.');
+    const prevGoVersionMajor = previousGoVersion[0];
+    const prevGoVersionMinor = previousGoVersion[1];
+    const prevGoVersionBugfix = previousGoVersion[2] || '';
+    console.error(prevGoVersionMajor, prevGoVersionMinor, prevGoVersionBugfix);
     await modifyRepoFile('workers/generic-worker/gw-decision-task/tasks.yml',
       contents => contents.replace(
-        /go [0-9]+\.[0-9]+\.[0-9]+/g,
-        `go ${goVersionMajor}.${goVersionMinor}.${goVersionBugfix}`,
+        /go [0-9.]+/g,
```

Comment:
I believe it would have same problem as regexp above - it would also match `go.sum` and `go.mod` strings